### PR TITLE
Fix issues with VA_INCLUDE_HEADERS when building with CUDA support

### PR DESCRIPTION
**Pull request by dmatveev**
### Issue description

* For some reason, `WITH_` is used instead of `HAVE_` when checking for VA libs in G-API CMakeLists
* `WITH_VA` is `TRUE` by default on x86 (see `opencv/CMakeLists.txt:414`)
* When configuring samples, the test for VA haven't been made (but `WITH_VA` is still true and it is ok)
* Removed redundant call to `OpenCVFindVA.cmake` as it is done anyway (see `opencv/CMakeLists.txt:743` and `opencv/cmake/OpenCVFindLibsVideo.cmake`)
* Removed FATAL ERROR case from G-API's CMakeLists.txt if VA isn't available. Have no idea why it is there

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Review by SergeyIvanov87 — APPROVED** [review]
LGTM

File: modules/gapi/CMakeLists.txt [hunk 1]
```diff
@@ -328,18 +328,11 @@ if(HAVE_GAPI_ONEVPL)
...
-        ocv_target_include_directories(opencv_test_gapi SYSTEM PRIVATE ${VA_INCLUDE_DIR})
-        ocv_target_link_libraries(${the_module} PRIVATE ${VA_LIBRARIES})
-        ocv_target_link_libraries(opencv_test_gapi PRIVATE ${VA_LIBRARIES})
-      endif()
-    else(WITH_VA)
-      message(FATAL_ERROR "libva not found: building HAVE_GAPI_ONEVPL without libVA support is impossible on UNIX systems")
-    endif(WITH_VA)
+  if(UNIX AND HAVE_VA)
+    ocv_target_include_directories(${the_module} SYSTEM PRIVATE ${VA_INCLUDE_DIR})
+    ocv_target_include_directories(opencv_test_gapi SYSTEM PRIVATE ${VA_INCLUDE_DIR})
```
Comment (alalek) [inline modules/gapi/CMakeLists.txt:333]: > opencv_test_gapi

Test target should be guarded, see code below.

Comment (dmatveev) [reply modules/gapi/CMakeLists.txt:333]: Good point, updated. Thanks

**Comment by dmatveev** [comment]
Also tested the build with VA being available:

```
root@7973346b3aad:~/opencv/build# ldd bin/example_gapi_onevpl_infer_single_roi | grep va
        libva.so.2 => /lib/x86_64-linux-gnu/libva.so.2 (0x00007fc3af609000)
        libva-drm.so.2 => /lib/x86_64-linux-gnu/libva-drm.so.2 (0x00007fc3af604000)
```

At least it builds now

**Comment by mshabunin** [comment]
> For some reason, WITH_ is used instead of HAVE_ when checking for VA libs in G-API CMakeLists
> Removed redundant call to OpenCVFindVA.cmake as it is done anyway

Perhaps it was designed like this for standalone G-API build. Do we still support this configuration?

**Comment by dmatveev** [comment]
@mshabunin yes we do, but VPL & VAAPI were never a part of it. Good question if it is still tested on the CI though..

**Pull request merged**